controllers/component_build_controller_simple_build.go

```diff
@@ -102,6 +67,18 @@ func (r *ComponentBuildReconciler) SubmitNewBuild(ctx context.Context, component
 		log.Error(err, "error getting git provider credentials secret", l.Action, l.ActionView)
 	}
 
+	// Try to find git secret in case private git repository is used
+	gitSecretName, err := r.findGitSecretName(ctx, component)
+	if err != nil {
+		log.Error(err, "failed to find git secret name")
+		return err
+	}
+	if pRunGitInfo == nil {
+		pRunGitInfo = &pipelineRunGitInfo{}
```

Comment:
when gitInfo is nil and gitSecretName is empty then it should fail with error on component. So that HAC could trigger OAuth workflow and retry.